sci-geosciences/qgis: install full set of icons and MIME types

cjmayo opened the pull request:
@Amynka - all the time there was a full set of icons and MIME database file under `debian/` ...

I think there is a problem with the application icon that is displayed - it is always 60x60 scaled. I've included a patch for that, but perhaps we should wait and see how that does upstream? https://github.com/qgis/QGIS/pull/3933

Also I picked a range of icon sizes, there are even more. Happy to add some.

SoapGentoo commented on sci-geosciences/qgis/qgis-2.18.2-r3.ebuild:
```diff
@@ -0,0 +1,198 @@
+IUSE="examples georeferencer grass mapserver oracle postgres python"
+
+REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} )
+		mapserver? ( python )"
```
too much indentation

cjmayo replied:
Changed to match the other multiline definitions.

SoapGentoo commented on sci-geosciences/qgis/qgis-2.18.2-r3.ebuild:
```diff
@@ -0,0 +1,198 @@
+	sys-devel/bison
+	sys-devel/flex"
+
+DOCS=( BUGS ChangeLog NEWS )
```
don't need these, part of default set already

cjmayo replied:
Removed. Gives us README.md too.

SoapGentoo commented on sci-geosciences/qgis/qgis-2.18.2-r3.ebuild:
```diff
@@ -0,0 +1,198 @@
+}
+
+src_prepare() {
+	default
```
don't use default with cmake! use `cmake-utils_src_prepare` instead

cjmayo replied:
Done.

SoapGentoo commented on sci-geosciences/qgis/qgis-2.18.2-r3.ebuild:
```diff
@@ -0,0 +1,198 @@
+			)
+		else
+			mycmakeargs+=(
+				"-DQWT_INCLUDE_DIR=/usr/include/qwt6"
```
is `-DQWT_INCLUDE_DIR` different between the two clauses? Just pull it out of the conditional and add it to `mycmakeargs` before the `if`

cjmayo replied:
Done.

SoapGentoo commented on sci-geosciences/qgis/qgis-2.18.2-r3.ebuild:
```diff
@@ -0,0 +1,198 @@
+	domenu debian/qgis.desktop
+
+	local types="qgs qml qlr qpt"
+	local sizes="16 22 24 32 48 96 128 256"
```
why do you need to define these? just put them in the loop directly and instead make the loop counters `local`

cjmayo replied:
I quite like having them all obvious up front, but agree it is wasteful. Removed and made loop counters local.

SoapGentoo commented on sci-geosciences/qgis/qgis-2.18.2-r3.ebuild:
```diff
@@ -0,0 +1,198 @@
+
+	if use examples; then
+		insinto /usr/share/doc/${PF}/examples
+		doins -r "${WORKDIR}"/qgis_sample_data/*
```
```
docinto examples
dodoc -r "${WORKDIR}"/qgis_sample_data/.
```
(notice the dot)

cjmayo replied:
Done.
I've also added a `docompress -x`. I think if you actually want to use these files it will be a pain if they are individually compressed. And you only get them if you set the USE flag.

SoapGentoo commented on sci-geosciences/qgis/qgis-2.18.2-r3.ebuild:
```diff
@@ -0,0 +1,198 @@
+		doins -r "${WORKDIR}"/qgis_sample_data/*
+	fi
+
+	python_optimize "${D}"/usr/share/qgis/python \
```
could you consider installing these? Install python modules into `/usr/share` is iffy

cjmayo replied:
It would be nice. But I think quite a few things to check to do it properly e.g.:
* Can they be replaced by packages from the tree?
* Have qgis modified them?
Unfortunately I haven't got the time at the moment and qgis isn't the only offender in the tree.

SoapGentoo commented on sci-geosciences/qgis/qgis-2.18.2-r3.ebuild:
```diff
@@ -0,0 +1,198 @@
+
+	if use grass; then
+		python_fix_shebang "${D}"/usr/share/qgis/grass/scripts
+		python_optimize "${D}"/usr/share/qgis/grass/scripts
```
same here, only python modules need to be byte-compiled, not scripts

cjmayo replied:
Agreed and done. I just saw /usr/share and assumed they were modules, the last part of the path should have been a clue.

cjmayo commented:
Also:
* Added the 64 pixel icons, it is one of the sizes mentioned in:
https://developer.gnome.org/gtk3/stable/GtkWindow.html#gtk-window-set-icon-list
* Removed `$(python_get_sitedir)/qgis` and `$(python_get_sitedir)/pyspatialite` from python_optimize. Alas we're not now installing anything in site-packages (see above).

Amynka commented:
Did you read the email about sip dependency? What is your opinion on it?

cjmayo commented:
I think it's only a real problem when sip-4.18 drops out of the tree. It's clearly a pain if you want to use the latest PyQt and the like.
I'm sure they won't miss this upstream.
Obviously best for it to be reported upstream and fixed there. I haven't got time right now - and I wouldn't want to distract them from my two outstanding PRs...
Also worth someone else checking it wasn't just me having the problem.

N.B. from the frontpage of qgis.org looks like maybe another release at the end of next week.

cjmayo commented:
I don't think the patch is necessary. I must have had something wrong. It is removed.

Therefore I believe this ebuild is good - subject to any further review.
May still want to hold off for a week in case 2.18.3 is released - the only noticeable changes are the icons. Unless you were hoping to stabilise 2.18.2 faster, then I suggest this is the 2.18.2 ebuild to pick.

Amynka commented:
So this PR is not necessary? I am confused :)

cjmayo commented:
Sorry to cause confusion, I'll try and do better:

The patch in FILESDIR is not needed. I've removed all traces of that.

The PR is needed.

By all means merge it - looks like the review needs to be approved.

If you haven't merged it by the time qgis 2.18.3 is released I'll update this PR with 2.18.3.
